### PR TITLE
fix: [CDS-95099]: Improvement of GitEx doc

### PR DESCRIPTION
**Pull request by krishi0408**
Thanks for contributing to the Harness Developer Hub! Our code owners will review your submission.

## Description

* Please describe your changes: Improvement of Git Experience doc 
* Jira/GitHub Issue numbers (if any): CDS-95099
* Preview links/images (Internal contributors only): \__________________

## PR lifecycle

We aim to merge PRs within one week or less, but delays happen sometimes.

If your PR is open longer than two weeks without any human activity, please tag a [code owner](https://github.com/harness/developer-hub/blob/main/.github/CODEOWNERS) in a comment.

PRs must meet these requirements to be merged:

- [x] Successful preview build.
- [ ] Code owner review.
- [x] No merge conflicts.
- [ ] Release notes/new features docs: Feature/version released to at least one prod environment.

### CODE REVIEW

File: docs/platform/git-experience/manage-environments-infra-definitions.md [hunk 1]
```diff
@@ -74,7 +74,8 @@ When adding a remote environment to the stage, you have the ability to choose a
 1. When the remote environment is linked to the pipeline, the environment branch is displayed as `__DEFAULT__`.
 `__DEFAULT__` resolves to the same branch as the pipeline, when both the environment and pipeline where it is linked are in the same repository or the main branch of the repository where the environment resides when it is different.
 2. Infrastructure definition takes the branch of the environment when both reside in the same repository.
```
Comment (roshnisarangadharan) [inline docs/platform/git-experience/manage-environments-infra-definitions.md:76]:
```suggestion
* When a remote environment is linked to a pipeline, the environment branch is displayed as `__DEFAULT__`.
`__DEFAULT__` resolves to the same branch as the pipeline when the pipeline and the environment linked to the pipeline are in the same repository. If the pipeline resides in a different repository, then _DEFAULT_ resolves to the main branch of the repository where the linked environment resides.
* Infrastructure definition takes the same branch of the environment when both reside in the same repository.
```

File: docs/platform/git-experience/manage-environments-infra-definitions.md [hunk 2]
```diff
@@ -74,7 +74,8 @@ When adding a remote environment to the stage, you have the ability to choose a
...
-3. When a remote stage template containing a remote environment is linked in a Remote Pipeline, the Stage template branch is defaulted when both entities are in the same repository. If the environment is in a different repository, that repository is taken. Read more about creating [remote stage template](/docs/platform/templates/create-a-remote-stage-template.md).
+3. If you're using a [remote stage template](/docs/platform/templates/create-a-remote-stage-template.md) along with a remote environment in a Remote Pipeline and both are in the same repository, the template's branch will be used by default.
```
Comment (roshnisarangadharan) [inline docs/platform/git-experience/manage-environments-infra-definitions.md:77]:
```suggestion
* If you're using a [remote stage template](/docs/platform/templates/create-a-remote-stage-template.md) along with a remote environment in a remote pipeline, and both are in the same repository, then the default branch is the branch where the template resides.
```

File: docs/platform/git-experience/manage-environments-infra-definitions.md [hunk 3]
```diff
@@ -74,7 +74,8 @@ When adding a remote environment to the stage, you have the ability to choose a
...
+4. However, if the environment is located in a different repository, the repository where the environment resides will be selected instead.
```
Comment (roshnisarangadharan) [inline docs/platform/git-experience/manage-environments-infra-definitions.md:78]:
```suggestion

```

Comment (roshnisarangadharan) [reply docs/platform/git-experience/manage-environments-infra-definitions.md:78]: Do not use numbered list if it is not a procedure. It looks like different pieces of info, and not a work flow. Use unordered/bulleted list in such cases.

File: docs/platform/git-experience/manage-services-using-git-experience.md [hunk 4]
```diff
@@ -75,12 +75,11 @@ To add a remote service to a pipeline, do the following:
...
 
-:::note
-
+:::info note
 1. When the remote Service is linked to the pipeline, the branch of the Service is displayed as `__DEFAULT__`.
 `__DEFAULT__` resolves to the same branch as the pipeline, when both the service and pipeline where it is linked are in the same repository or the main branch of the repository where the Service resides when it is different.
-2. When a remote stage template containing remote service is linked in a Remote pipeline, the pipeline's branch is automatically set to default when both entities are in the same repository. If a service is located in a different repository, the `main` of the service repository is used as the default. For more information, go to [create a remote stage template](/docs/platform/templates/create-a-remote-stage-template.md).
-
+2. If a [remote stage template](/docs/platform/templates/create-a-remote-stage-template.md) with a remote service is connected in a Remote Pipeline and both are in the same repository, the pipeline's branch defaults to the main one.
+3. However, if the service is in a different repository, the main branch of the service's repository is automatically chosen as the default.
```
Comment (roshnisarangadharan) [inline docs/platform/git-experience/manage-services-using-git-experience.md:82]: Apply same comments from the previous file.

File: docs/platform/git-experience/manage-environments-infra-definitions.md (hunk 1 above)
Comment (roshnisarangadharan) [reply docs/platform/git-experience/manage-environments-infra-definitions.md]: Please check the technical accuracy of the suggestion above.

File: docs/platform/git-experience/manage-services-using-git-experience.md (hunk 4 above)
Comment (krishi0408) [reply docs/platform/git-experience/manage-services-using-git-experience.md]: done thanks for the review @roshnisarangadharan

File: docs/platform/git-experience/manage-services-using-git-experience.md [hunk 5]
```diff
@@ -78,8 +78,7 @@ To add a remote service to a pipeline, do the following:
 :::info note
 1. When the remote Service is linked to the pipeline, the branch of the Service is displayed as `__DEFAULT__`.
 `__DEFAULT__` resolves to the same branch as the pipeline, when both the service and pipeline where it is linked are in the same repository or the main branch of the repository where the Service resides when it is different.
-2. If a [remote stage template](/docs/platform/templates/create-a-remote-stage-template.md) with a remote service is connected in a Remote Pipeline and both are in the same repository, the pipeline's branch defaults to the main one.
-3. However, if the service is in a different repository, the main branch of the service's repository is automatically chosen as the default.
+2. If a [remote stage template](/docs/platform/templates/create-a-remote-stage-template.md) with a remote service is connected in a Remote Pipeline and both are in the same repository, the pipeline's branch defaults to the main one.However, if the service is in a different repository, the main branch of the service's repository is automatically chosen as the default.
```
Comment (roshnisarangadharan) [inline docs/platform/git-experience/manage-services-using-git-experience.md:81]:
```suggestion
2. If a [remote stage template](/docs/platform/templates/create-a-remote-stage-template.md) with a remote service is connected in a Remote Pipeline and both are in the same repository, the pipeline's branch defaults to the main one. However, if the service is in a different repository, the main branch of the service's repository is automatically chosen as the default.
```

File: docs/platform/git-experience/manage-services-using-git-experience.md [hunk 6]
```diff
@@ -78,8 +78,7 @@ To add a remote service to a pipeline, do the following:
 :::info note
 1. When the remote Service is linked to the pipeline, the branch of the Service is displayed as `__DEFAULT__`.
```
Comment (roshnisarangadharan) [inline docs/platform/git-experience/manage-services-using-git-experience.md:79]:
```suggestion
* When the remote service is linked to the pipeline, the branch of the service is displayed as `__DEFAULT__`.
```

File: docs/platform/git-experience/manage-services-using-git-experience.md [hunk 7]
```diff
@@ -78,8 +78,7 @@ To add a remote service to a pipeline, do the following:
...
 `__DEFAULT__` resolves to the same branch as the pipeline, when both the service and pipeline where it is linked are in the same repository or the main branch of the repository where the Service resides when it is different.
```
Comment (roshnisarangadharan) [inline docs/platform/git-experience/manage-services-using-git-experience.md:80]:
```suggestion
`__DEFAULT__` resolves to the same branch as the pipeline when both the service linked to the pipeline are in the same repository. If the pipeline resides in a different repository, then _DEFAULT_ resolved to the main branch of the repository where the linked service resides.
```

**Review by shylajasun — COMMENTED** [review]
@krishi0408 Could you fix the corrections as discussed

**Pull request merged**